mpi-proxy-split/test/run_mana_mpi_regression_test.py

```diff
@@ -0,0 +1,202 @@
+                return 1
+            eprint(verbose,"Checkpoint command successful")
+
+            # Wait for checkpoint to complete, make sure no ranks die
+            while get_running_ranks(mana_dir, "WorkerState::RUNNING") \
+                    != int(ranks):
+                if get_running_ranks(mana_dir) != int(ranks):
+                    eprint(verbose,"Test case failed: not enough \
+                        ranks running while checkpointing")
+                    return 6
+
+            # Kill DMTCP Coord
+            kill_child = subprocess.Popen(['pkill', '-9', 'dmtcp_coord'])
+            kill_child.wait()
+
+            # Wait for DMTCP Coord to die, then restart it
+            status = subprocess.run([f'{mana_dir}/bin/mana_status'],
+                                     stdout=PIPE, stderr=subprocess.DEVNULL)
+            while "Checking for coordinator" \
+                    not in status.stdout.decode('utf-8'):
+                status = \
+                    subprocess.run([f'{mana_dir}/bin/mana_status'],
+                                    stdout=PIPE, stderr=PIPE)
+            eprint(verbose,"Checkpointed, restarting...")
+            subprocess.run([f"{mana_dir}/bin/mana_coordinator"],
+                           stderr=subprocess.DEVNULL, stdout=subprocess.DEVNULL)
+
+            # Wait for coord to restart, then restart ranks
+            time.sleep(1)
+            run_child = subprocess.Popen([f'{restart}', '-n', f'{ranks}',
+                                         f'{mana_dir}/bin/mana_restart'],
+                                         stdout=subprocess.DEVNULL,
+                                         stderr=subprocess.DEVNULL)
+
+            # Wait for ranks to restart without dying
+            ranks_seen = 0
+            while get_running_ranks(mana_dir, "WorkerState::RUNNING") \
+                    != int(ranks):
+                if get_running_ranks(mana_dir) < ranks_seen:
+                    eprint(verbose,"Test case failed: not enough \
+                        ranks running after restart")
+                    return 2
+                ranks_seen = get_running_ranks(mana_dir)
+            time.sleep(3)
+
+            clean()
+        else: # Child died earlier, check retval to see if successful
+            if run_child.poll() != 0:
+                eprint(verbose,"Test case failed: child exited early")
+                return 3
+            else:
+                eprint(verbose,"Child exited early: increase iterations")
+                return 0
+
+    # Check enough ranks running at exit, then return
+    # Kills mana_coordinator + spawned MPI children
+    if get_running_ranks(mana_dir) != int(ranks):
+        eprint(verbose,"Test case failed: not enough "
+               "ranks running when exiting")
+        run_child.kill()
+        return 4
+    else:
+        eprint(verbose,"Test case passed")
+        run_child.kill()
+        return 0
+
+    run_child.kill()
```

Comment:
In all cases, your code is returning! Your script will not reach to this code. I guess you can safely remove this line. Otherwise, you can have a comment stating that the workflow won't reach here normally, but kill if it reaches here and return an error code.